src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlTools.SqlCore.Connection;
+using Microsoft.SqlTools.SqlCore.ObjectExplorer;
+using Microsoft.SqlTools.SqlCore.ObjectExplorer.Nodes;
+using Microsoft.SqlTools.SqlCore.ObjectExplorer.SmoModel;
+
+namespace Microsoft.SqlTools.CoreSql.ObjectExplorer
+{
+    /// <summary>
+    /// Stateless object explorer class can be used to handle object explorer requests without creating a session. It requires a connection string and a node path to query objects from the server.
+    /// </summary>
+    public class StatelessObjectExplorer
+    {
+        /// <summary>
+        /// Expands the node at the given path and returns the child nodes.
+        /// </summary>
+        /// <param name="connectionString"> Connection string to connect to the server </param>
+        /// <param name="accessToken"> Access token to connect to the server. To be used in case of AAD based connections </param>
+        /// <param name="nodePath"> Path of the node to expand </param>
+        /// <param name="serverInfo"> Server information </param>
+        /// <param name="options"> Object explorer options </param>
+        /// <param name="filters"> Filters to be applied on the leaf nodes </param>
+        /// <returns> Array of child nodes </returns>
+        /// <exception cref="ArgumentNullException"> Thrown when the parent node is not found </exception>
+        /// <exception cref="TimeoutException"> Thrown when the operation times out.</exception> <summary>
+        /// </summary>     
+        public static TreeNode[] Expand(string connectionString, SecurityToken? accessToken, string nodePath, ObjectExplorerServerInfo serverInfo, ObjectExplorerOptions options, INodeFilter[]? filters = null)
```

Comment:
What about just having this as a static method on the OE class? Seems better to keep logic all in one place that's related to OE.

And make sure you're trying to consolidate as much logic as you can, it would be great not to have to remember to fix stuff in two separate places that aren't directly related.

Reply:
I agree. For now, I am removing OE class and just keeping this stateless OE class in the package. OE class would require more implementation to completely replace the code we use in OE service (in service layer). However, due to time constraints, I need to get something working out.